src/app/application/components/route-view/README.md

```diff
@@ -8,23 +8,107 @@ routable `*View.vue` component**. Think of it as your `<html>` tag.
 
 ::: danger
 When using this you should specify the name of the route you are creating as
-the RouteView's `name` property.
+the RouteView's `name` property. See `name="route-name"` in the below example.
 :::
 
-`RouteView` contains functionality to:
+`RouteView` contains functionality for:
 
-- make it easy to set top-level DOM values that you usually wouldn't have
-  direct access to, for example [setting the HTML `<title>`](#setting-the-title-of-the-page) or [the HTML `class` attribute](#setting-html-node-attributes).
-- [provide access to utilities/tooling commonly used in route views](#exposing-commonly-used-utilities) i.e.
-  `route`, `t` and `uri`
+- ["Better" `route.params`](#better-route-params)
+  - define and access reactive route params with a finer level of granularity
+    to using `vue-router`s `route`
+  - specify type and provide defaults for route params
+  - automatically transform boolean params to boolean query params.
+  - provide route params as a two-way binding
+- setting top-level DOM values
+  - [the HTML `<title>`](#setting-the-title-of-the-page)
+  - [the HTML `class` attribute](#setting-root-node-attributes).
+- [providing access to utilities/tooling commonly used in route views](#exposing-commonly-used-utilities)
+  such as `t`, `can`, `me`
+
+It generally looks something like this:
+
+```vue
+<RouteView
+  name="route-name"
+  :params="{
+    define: ''
+    route: ''
+    params: false
+  }"
+  v-slot="{ route, t, can, me }"
+>
+...View contents...
+</RouteView>
+
+```
+
+## "Better" route params
+
+::: danger
+`route` is not an instance of `vue-router`'s `route`. It is very similar but
+omits some things and adds functionality to others.
+:::
+
+```vue
+<RouteView
+  name="route-name"
+  :params="{
+    mesh: ''
+    service: ''
+    page: 1
+    checked: false
+  }"
+  v-slot="{ route }"
+>
+  Page: {{ route.params.page }}
+  ...
+</RouteView>
+```
+
+The above example illustrates how to define the dependencies of your route
+template/view, also known as the route params. If you do not need a specific
+route param in a template you should not define them in this specific
+template/RouteView.
+
+::: warning
+Note: at this level there is no concept of "path vs query" params. They are
+just all "params". Whether they end up being path or query params is controlled
+by your routing config. If you refer to the param in a URL in your route
+config, they end up being path params, if not, they are query params.
+:::
+
+
+Defining these "dependencies" here has several advantages:
+
+1. Accessing `route.params` gives you autocompletion/type-checking in your IDE
```

Comment:
Kind of a side thought here 🤔 

When looking at `vue-router/typed-routes` just now, it uses typescript to define the type of the URL parameter. The interesting thing is URL params are always strings, you need something at runtime to convert something you say should be treated as a number (or boolean) to be that type. I wonder if/how `vue-router/typed-routes` converts the URL string to the type you are expecting correctly at runtime by using the build time type definitions.

Reply:
Sorry one more thing thats vaguely related here, and now is probably a good time to mention in a food for thought kind of way.

I would really like to use `https://github.com/posva/unplugin-vue-router` to automatically create our routing config from the filesystem (and therefore nest our *View.vue files instead of the flat structure they currently have) I did try it out a while back but didn't manage to get it to work.

Would be good to try it again at some point, or someone with a different pair of eyes might have more success at getting that to work.

If using `https://github.com/posva/unplugin-vue-router` that sounds interesting to you we should discuss as a group and see what comes of it.

Reply:
> but even if we did, it wouldn't give us the default params, plus we'd still have to define the params we use in the template

fully agree, would be awesome to keep that, but would be nice to get the typesafety on top (I hope I didn't miss anything in that regards 😄 )

>When looking at vue-router/typed-routes just now, it uses typescript to define the type of the URL parameter. The interesting thing is, URL params are always strings i.e. a URL is a string, you need something at runtime to convert something you say should be treated as a number (or boolean) to be that type. I wonder if/how vue-router/typed-routes converts the URL string to the type you are expecting correctly at runtime by using the build time type definitions.

I think we'd have to do the conversion ourselves when needed 🙂 

>I would really like to use https://github.com/posva/unplugin-vue-router to automatically create our routing config from the filesystem (and therefore nest our *View.vue files instead of the flat structure they currently have) I did try it out a while back but didn't manage to get it to work.
> Would be good to try it again at some point, or someone with a different pair of eyes might have more success at getting that to work.
> If using https://github.com/posva/unplugin-vue-router that sounds interesting to you we should discuss as a group and see what comes of it.

I am definitely open for that 🙂 but if I understand it correctly, the param types would also be always string and we'd have to convert them when needed 🤔